cmd/microcloud/ask.go

```diff
@@ -1104,6 +1104,83 @@ func (c *initConfig) askOVNNetwork(sh *service.Handler) error {
 		}
 	}
 
+	canOVNUnderlay := true
+	for peer, state := range c.state {
+		if len(state.AvailableOVNInterfaces) == 0 {
+			fmt.Printf("Not enough interfaces available on %s to create an underlay network, skipping\n", peer)
+			canOVNUnderlay = false
+			break
+		}
+	}
+
+	var ovnUnderlaySelectedIPs map[string]string
+	if canOVNUnderlay {
+		// TODO: Call `sh.Services[types.MicroOVN].(*service.OVNService).SupportsFeature(context.Background(), "custom_encapsulation_ip")`
+		// when MicroCloud will be updated with microcluster/v2
+
+		wantsDedicatedUnderlay, err := c.asker.AskBool("Configure dedicated underlay networking? (yes/no) [default=no]: ", "no")
+		if err != nil {
+			return err
+		}
+
+		if wantsDedicatedUnderlay {
+			header = []string{"LOCATION", "IFACE", "TYPE", "IP ADDRESS (CIDR)"}
+			fmt.Println("Select exactly one network interface from each cluster member:")
+			data = [][]string{}
+
+			for peer := range c.systems {
+				// skip any systems that have already been clustered, but are available for other configuration.
+				state, ok := c.state[c.name]
+				if ok {
+					if state.ExistingServices[types.MicroOVN][peer] != "" {
+						continue
+					}
+				}
+
+				for _, net := range state.AvailableOVNInterfaces {
```

Comment:
You're iterating on the variable `state` which you instantiate above to always point to `c.state[c.name]` so all entries in the table are getting the same address.

You need `c.systems[peer].AvailableOVNInterfaces`, for this loop, not `state.AvailableOVNInterfaces`.

Reply:
fixed